lib/taskana-core/src/main/java/pro/taskana/task/internal/AttachmentHandler.java

```diff
@@ -76,18 +78,69 @@
   }
 
   void insertAndDeleteAttachmentsOnTaskUpdate(TaskImpl newTaskImpl, TaskImpl oldTaskImpl)
-      throws AttachmentPersistenceException {
-
+      throws AttachmentPersistenceException, InvalidArgumentException,
+          ClassificationNotFoundException {
     List<Attachment> newAttachments =
         newTaskImpl.getAttachments().stream().filter(Objects::nonNull).collect(Collectors.toList());
     newTaskImpl.setAttachments(newAttachments);
 
+    for (Attachment attachment : newAttachments) {
+      verifyAttachment((AttachmentImpl) attachment);
+    }
+
     deleteRemovedAttachmentsOnTaskUpdate(newTaskImpl, oldTaskImpl);
     insertNewAttachmentsOnTaskUpdate(newTaskImpl, oldTaskImpl);
     updateModifiedAttachmentsOnTaskUpdate(newTaskImpl, oldTaskImpl);
   }
 
-  void updateModifiedAttachmentsOnTaskUpdate(TaskImpl newTaskImpl, TaskImpl oldTaskImpl) {
+  void insertNewAttachmentsOnTaskCreation(TaskImpl task)
+      throws InvalidArgumentException, AttachmentPersistenceException,
+          ClassificationNotFoundException {
+    List<Attachment> attachments = task.getAttachments();
+
+    if (attachments != null) {
+      for (Attachment attachment : attachments) {
+        AttachmentImpl attachmentImpl = (AttachmentImpl) attachment;
+        verifyAttachment(attachmentImpl);
+        initAttachment(attachmentImpl, task);
+
+        try {
+          attachmentMapper.insert(attachmentImpl);
+          if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug(
+                "TaskService.updateTask() for TaskId={} INSERTED an Attachment={}.",
+                task.getId(),
+                attachmentImpl);
+          }
+        } catch (PersistenceException e) {
+          throw new AttachmentPersistenceException(
+              String.format(
+                  "Cannot insert the Attachement %s for Task %s  because it already exists.",
+                  attachmentImpl.getId(), task.getId()),
+              e.getCause());
+        }
+      }
+    }
+  }
+
+  private void insertNewAttachmentsOnTaskUpdate(TaskImpl newTaskImpl, TaskImpl oldTaskImpl)
+      throws AttachmentPersistenceException, ClassificationNotFoundException {
+    List<String> oldAttachmentIds =
+        oldTaskImpl.getAttachments().stream()
+            .map(AttachmentSummary::getId)
+            .collect(Collectors.toList());
+
+    List<Attachment> newAttachments =
```

Comment:
should we maybe abstract this filtering of the new Attachments into a separate method?

Reply:
I was thinking about line 128 until 136